api/campaigns/class-maybe-show-campaign.php

```diff
@@ -200,6 +200,16 @@ public function should_campaign_be_shown( $client_id, $campaign, $settings, $ref
 			if ( $campaign_segment->is_not_donor && $is_donor ) {
 				$should_display = false;
 			}
+			if ( isset( $campaign_segment->referrers ) && $campaign_segment->referrers && ! empty( $campaign_segment->referrers ) && isset( $_REQUEST['ref'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+				$referer_domain   = parse_url( $_REQUEST['ref'], PHP_URL_HOST ); // phpcs:ignore WordPress.WP.AlternativeFunctions.parse_url_parse_url, WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+				$referrer_matches = in_array(
+					$referer_domain,
+					explode( ',', $campaign_segment->referrers )
```

Comment:
```suggestion
					array_map( 'trim', explode( ',', $campaign_segment->referrers ) )
```

Fixes issue if there are spaces around the commas, e.g. `google.com, facebook.com`.